packages/stencil-library/src/components/button/button.tsx

```diff
@@ -25,57 +21,87 @@ export class Button {
-                case 'danger':
-                    return "usa-button--secondary"
             }
         }
 
         const getDarkModeVariantClassName = (variant: ButtonVariant) => {
             switch (variant) {
+                case 'primary':
+                    return "primary-button-dark"
                 case "secondary":
                     return "usa-button--outline usa-button--inverse"
                 case 'link':
-                    return "usa-button--unstyled usa-button--outline usa-button--inverse"
+                    return "usa-button--unstyled unstyled-button-dark"
+                default:
+                    return 'primary-button-dark'
+            }
+        }
+
+        const getDangerModeVariantClassName = (variant: ButtonVariant) => {
+            switch (variant) {
+                case 'primary':
+                    return "usa-button--secondary"
+                case "secondary":
+                    return "usa-button--outline outline-danger"
+                case 'link':
+                    return "usa-button--unstyled unstyled-button-danger"
                 default:
                     return getVariantClassName(variant)
             }
         }
 
-        return this.mode === "light"
-            ? `usa-button ${getVariantClassName(this.variant)}`
-            : `usa-button ${getDarkModeVariantClassName(this.variant)}`
+        let getClassName;
+        switch (this.mode) {
+            case "light":
+                getClassName = getVariantClassName;
+                break;
+            case "dark":
+                getClassName = getDarkModeVariantClassName;
+                break;
+            case "danger":
+                getClassName = getDangerModeVariantClassName;
+                break;
+            default:
+                getClassName = getVariantClassName;
+        }
+
+        return `usa-button ${getClassName(this.variant)} ${this.icon ? 'button-icon' : ''}`
     }
 
-    componentWillLoad() {
-        if (this.asChild) {
-            const slotElements = this.hostElement.children
-            if (slotElements.length !== 1) {
-                throw new Error(`If the asChild property is set to true on the njwds-button component, the component must have exactly one slot element. Instead got ${slotElements.length} elements.`)
-            }
-            if (slotElements[0].tagName !== "BUTTON") {
-                throw new Error(`If the asChild property is set to true on the njwds-button component, the slot element must be a <button>. Instead got ${slotElements[0].outerHTML}`)
-            }
-            const buttonElement = slotElements[0] as HTMLButtonElement
-            const buttonClassName = this.getButtonClassName()
-            buttonElement.className = `${buttonElement.className} ${buttonClassName}`
+    private renderIcon() {
+        let iconClass = ''
+        switch (this.iconPosition) {
+            case "leading":
+                iconClass += ' margin-right-105'
```

Comment:
[Dust] Why `+=` vs. just setting it directly? Since the initial class is empty

Reply:
Good call, originally I had other text in iconClass but I took it out